cypress/e2e/SubdomainNavigation.cy.ts

```diff
@@ -0,0 +1,221 @@
+				const expectedText = index === 0 ? 'wilder' : 'snowflake';
+
+				cy.wrap(a)
+					.should('have.attr', 'href', expectedHref)
+					.should('have.text', expectedText);
+			});
+
+		// assert expected elements
+		cy.assertElementIsVisible(
+			SubdomainViewSelector.VIEW_SUBDOMAINS_BANNER_CONTAINER,
+		);
+		cy.assertElementIsVisible(
+			SubdomainViewSelector.VIEW_SUBDOMAINS_DETAILS_CARD_CONTAINER,
+		).and('contain', 'Winter Time');
+	});
+
+	it('can view grid view and go to corret domain', () => {
+		cy.visitRootDomain();
+
+		// assert expected section element
+		cy.assertElementIsVisible(SubdomainViewSelector.VIEW_SUBDOMAINS_SECTION);
+
+		// find toggle view radio group
+		cy.findByRole('group').should('be.visible');
+
+		// find toggle view radio buttons and assert attributes
+		cy.findAllByRole('radio')
+			.should('be.visible')
+			.should('have.length', 2)
+			.each(($radio, index) => {
+				if (index === 0) {
+					cy.wrap($radio)
+						.should('have.attr', 'data-state', 'off')
+						.should('have.attr', 'aria-checked', 'false');
+				} else if (index === 1) {
+					cy.wrap($radio)
+						.should('have.attr', 'data-state', 'on')
+						.should('have.attr', 'aria-checked', 'true');
+				}
+			});
+
+		// assert grid view
+		cy.get('[class*="Grid"]').should('be.visible');
+
+		// find and click sudomain grid card
+		cy.findByText(`0://${subdomainZna}`).should('exist').click();
+
+		// assert url
+		cy.assertUrl('contain', subdomainZna);
+	});
+
+	it('can view list view and go to corret domain', () => {
+		cy.visitRootDomain();
+
+		// assert expected section element
+		cy.assertElementIsVisible(SubdomainViewSelector.VIEW_SUBDOMAINS_SECTION);
+
+		// find toggle view radio group
+		cy.findByRole('group').should('be.visible');
+
+		// find list view radio button and click
+		cy.findAllByRole('radio').first().click();
+
+		// find toggle view radio buttons and assert attributes
+		cy.findAllByRole('radio')
+			.should('have.length', 2)
+			.each(($radio, index) => {
+				if (index === 0) {
+					cy.wrap($radio)
+						.should('have.attr', 'data-state', 'on')
+						.should('have.attr', 'aria-checked', 'true');
+				} else if (index === 1) {
+					cy.wrap($radio)
+						.should('have.attr', 'data-state', 'off')
+						.should('have.attr', 'aria-checked', 'false');
+				}
+			});
+
+		// assert list view
+		cy.assertElementIsVisible('table');
+
+		// find and click sudomain grid card
+		cy.findByText(`0://${subdomainZna}`).should('exist').click();
+
+		// assert url
+		cy.assertUrl('contain', subdomainZna);
+	});
+
+	it('can type in search bar to search by exact zNA (Grid)', () => {
+		cy.visitRootDomain();
+
+		// assert expected section element
+		cy.assertElementIsVisible(SubdomainViewSelector.VIEW_SUBDOMAINS_SECTION);
+
+		// find search bar input and enter zna
+		cy.findByPlaceholderText('Search by exact zNA')
+			.should('be.visible')
+			.type('snowflake');
```

Comment:
I wonder if we should move this test to zUI, and keep e2e for critical functionality? Not sure, will discuss!

Reply:
hmm yeah good point. I think this could still be relevant here as we're still following what steps the user would take to navigate to a domain via the search controls.